apache-cassandra/dashboards/cassandra-overview.libsonnet

```diff
@@ -0,0 +1,1782 @@
+      custom: {
+        fillOpacity: 5,
+        gradientMode: 'none',
+        hideFrom: {
+          legend: false,
+          tooltip: false,
+          viz: false,
+        },
+        lineWidth: 1,
+      },
+      mappings: [],
+      thresholds: {
+        mode: 'absolute',
+        steps: [
+          {
+            color: 'green',
+            value: null,
+          },
+          {
+            color: 'red',
+            value: 80,
+          },
+        ],
+      },
+      unit: 's',
+    },
+    overrides: [
+      {
+        matcher: {
+          id: 'byFrameRefID',
+          options: 'A',
+        },
+        properties: [
+          {
+            id: 'color',
+            value: {
+              fixedColor: 'red',
+              mode: 'fixed',
+            },
+          },
+        ],
+      },
+      {
+        matcher: {
+          id: 'byFrameRefID',
+          options: 'B',
+        },
+        properties: [
+          {
+            id: 'color',
+            value: {
+              fixedColor: 'blue',
+              mode: 'fixed',
+            },
+          },
+        ],
+      },
+      {
+        matcher: {
+          id: 'byFrameRefID',
+          options: 'C',
+        },
+        properties: [
+          {
+            id: 'color',
+            value: {
+              fixedColor: 'green',
+              mode: 'fixed',
+            },
+          },
+        ],
+      },
+      {
+        matcher: {
+          id: 'byFrameRefID',
+          options: 'D',
+        },
+        properties: [
+          {
+            id: 'color',
+            value: {
+              fixedColor: 'purple',
+              mode: 'fixed',
+            },
+          },
+        ],
+      },
+    ],
+  },
+  options: {
+    bucketOffset: 0,
+    combine: false,
+    legend: {
+      calcs: [],
+      displayMode: 'list',
+      placement: 'bottom',
+      showLegend: true,
+    },
+  },
+};
+
+local writeLatencyPanel(matcher) = {
+  datasource: promDatasource,
+  targets: [
+    prometheus.target(
+      'sum(cassandra_keyspace_writelatency_seconds{' + matcher + ', quantile="0.50"} >= 0) by (cluster)',
+      datasource=promDatasource,
+      legendFormat='{{ cluster }} - p50',
+    ),
+    prometheus.target(
+      'sum(cassandra_keyspace_writelatency_seconds{' + matcher + ', quantile="0.75"} >= 0) by (cluster)',
+      datasource=promDatasource,
+      legendFormat='{{ cluster }} - p75',
+    ),
+    prometheus.target(
+      'sum(cassandra_keyspace_writelatency_seconds{' + matcher + ', quantile="0.95"} >= 0) by (cluster)',
+      datasource=promDatasource,
+      legendFormat='{{ cluster }} - p95',
+    ),
+    prometheus.target(
+      'sum(cassandra_keyspace_writelatency_seconds{' + matcher + ', quantile="0.95"} >= 0) by (cluster)',
+      datasource=promDatasource,
+      legendFormat='{{ cluster }} - p99',
+    ),
+  ],
+  type: 'histogram',
+  title: 'Write latency',
+  description: 'Average local write latency for this cluster',
```

Comment:
I think there's been a misunderstanding here, and it's entirely on me 😅 
When we reviewed the spec, I recommended using a histogram, but didn't specify which visualisation style which I definitely should have!

What I was thinking of including is actually two panels, namely a latency histogram *heatmap*, and a Quartile Latency *time series* panel, using the histogram *data*. Again entirely on me, but to get an idea of what I'm thinking, here's a few pictures:
![image](https://user-images.githubusercontent.com/1282515/220901923-c6f4aeff-b285-41d0-9f7f-8e6a41bbfae9.png)
![image](https://user-images.githubusercontent.com/1282515/220901942-67d64f89-82d1-4747-85ba-5d850a547066.png)

Taken from this article: https://towardsdatascience.com/prometheus-histograms-with-grafana-heatmaps-d556c28612c7

@gloriasc would love to hear your thoughts on this as well :)

Reply:
Yes! Absolutely! 100% agreed! Those graphs would be awesome